### ISSUE TITLE
Bug (Improved Concentration Aura) (BB #610)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#610]
This issue was migrated from bitbucket.
**Original Reporter:** Flywithme
**Original Date:** 22.07.2014 14:05:06 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610

<hr>

Aura is not working properly "Improved Concentration Aura" http://www.wowwiki.com/Improved_Concentration_Aura,  For mage Counterspell,  shock shaman - works properly.  Bug is that it should also run on Pamela warriors 
http://www.wowwiki.com/Pummel and kick Rogues http://www.wowwiki.com/Kick

For example: if you use the talent, kick rouge reduced from 5 seconds to 3 seconds., Pummel warriors with 4 seconds to 2.5 seconds.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 22.07.2014 14:05:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11371327

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 22.07.2014 18:01:28 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11376606

<hr>

Could you please try after running this on your world database? Let me know how it goes :)

```
#!sql

UPDATE spell_affect SET SpellFamilyMask = 549591799 WHERE entry = 12536 AND effectId = 0;
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 23.07.2014 15:57:22 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11396320

<hr>

does not work(

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 23.07.2014 18:41:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11398878

<hr>

THANKS! :D

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 23.07.2014 18:59:24 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11399162

<hr>

Now try...

```
#!c++

# HG changeset patch
# User LordUsagi
# Date 1406141933 10800
#      Wed Jul 23 15:58:53 2014 -0300
# Node ID 534905cf84167aa34fd002fadd7c100fec0dfc41
# Parent  998d4712e9a97148325b1a26d94e268d964c78c2
[Core/Spells] Fix auras such as Improved Concentration that should reduce the
              time of certain debuffs. Also made m_currentSpells public.
              Solves OC Issue #610.

diff -r 998d4712e9a9 -r 534905cf8416 src/game/SpellEffects.cpp
--- a/src/game/SpellEffects.cpp Thu Jul 10 11:59:58 2014 -0300
+++ b/src/game/SpellEffects.cpp Wed Jul 23 15:58:53 2014 -0300
@@ -4515,7 +4515,7 @@
         {
             SpellEntry const* curSpellInfo = spell->m_spellInfo;
             // check if we can interrupt spell
-            if (spell->getState() == SPELL_STATE_CASTING || (spell->getState() == SPELL_STATE_PREPARING && spell->GetCastTime() > 0.0f))
+            if ((spell->getState() == SPELL_STATE_CASTING || (spell->getState() == SPELL_STATE_PREPARING && spell->GetCastTime() > 0.0f))  && unitTarget->m_currentSpells[i]->m_spellInfo->InterruptFlags & SPELL_INTERRUPT_FLAG_INTERRUPT && unitTarget->m_currentSpells[i]->m_spellInfo->PreventionType == SPELL_PREVENTION_TYPE_SILENCE)
             {
                 if ((i == CURRENT_GENERIC_SPELL && curSpellInfo->InterruptFlags & SPELL_INTERRUPT_FLAG_INTERRUPT) ||                   
                    (i == CURRENT_CHANNELED_SPELL && curSpellInfo->ChannelInterruptFlags & CHANNEL_FLAG_MOVEMENT))
diff -r 998d4712e9a9 -r 534905cf8416 src/game/Unit.h
--- a/src/game/Unit.h   Thu Jul 10 11:59:58 2014 -0300
+++ b/src/game/Unit.h   Wed Jul 23 15:58:53 2014 -0300
@@ -1336,6 +1336,8 @@
         Spell* FindCurrentSpellBySpellId(uint32 spell_id) const;
         int32 GetCurrentSpellCastTime(uint32 spell_id) const;

+        Spell* m_currentSpells[CURRENT_MAX_SPELL];
+
         uint32 m_addDmgOnce;
         uint64 m_SummonSlot[MAX_SUMMON_SLOT];
         uint64 m_ObjectSlot[4];
@@ -1668,8 +1670,6 @@
         uint32 m_CombatTimer;
         uint32 m_lastManaUse;                               // msecs

-        Spell* m_currentSpells[CURRENT_MAX_SPELL];
-
         UnitVisibility m_Visibility;

         Diminishing m_Diminishing;

```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 27.07.2014 03:57:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11454305

<hr>

Flywithme, you didn't let me know if this fixed your problem.

Please respond.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 27.07.2014 07:36:33 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11456239

<hr>

Today I will tell you, the problems were

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Flywithme
**Original Date:** 28.07.2014 17:57:24 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11479404

<hr>

does not work

---

If you have misunderstood the problem, please inquire)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 29.07.2014 23:53:43 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/610/#comment-11504628

<hr>

Resolved in commit 299b4d7e3890318b76187c0595a81af13bf68de2
